api/pkg/guard/guard.go

```diff
@@ -1,56 +1,129 @@
-// Package guard is a helper package to evaluate question about members in ShellHub.
+// Package guard is a helper to work around permissions on ShellHub API.
 package guard
 
 import (
-	"errors"
-
-	"github.com/shellhub-io/shellhub/pkg/authorizer"
 	"github.com/shellhub-io/shellhub/pkg/models"
 )
 
-var ErrForbidden = errors.New("resource forbidden")
+const (
+	RoleObserver      = "observer"
+	RoleOperator      = "operator"
+	RoleAdministrator = "administrator"
+	RoleOwner         = "owner"
+)
+
+const (
+	// RoleObserverCode is a role code for observer.
+	RoleObserverCode = iota + 1
+	// RoleOperatorCode is a role code for operator.
+	RoleOperatorCode
+	// RoleAdministratorCode is a role code for administrator.
+	RoleAdministratorCode
+	// RoleOwnerCode is a role code for owner.
+	RoleOwnerCode
+)
+
+// Roles maps all roles to its code.
+var Roles = map[string]int{
+	RoleObserver:      RoleObserverCode,
+	RoleOperator:      RoleOperatorCode,
+	RoleAdministrator: RoleAdministratorCode,
+	RoleOwner:         RoleOwnerCode,
+}
+
+// RolePermissions maps roles to its Permissions. It is used to check if a models.Member has permission to do something.
+var RolePermissions = map[string]Permissions{
+	RoleObserver:      observerPermissions,
+	RoleOperator:      operatorPermissions,
+	RoleAdministrator: adminPermissions,
+	RoleOwner:         ownerPermissions,
+}
 
-// CheckMember checks if a user is a namespace's member.
+// CheckMember checks if a models.User's ID is a models.Namespace's member. A models.User is a member if its ID is in
+// the models.Namespace's members list.
 func CheckMember(namespace *models.Namespace, id string) (*models.Member, bool) {
-	var memberFound models.Member
-	for _, memberSearch := range namespace.Members {
-		if memberSearch.ID == id {
-			memberFound = memberSearch
+	var found models.Member
+	for _, member := range namespace.Members {
+		if member.ID == id {
+			found = member
 
 			break
 		}
 	}
 
-	if memberFound.ID == "" || memberFound.Role == "" {
+	if found.ID == "" || found.Role == "" {
 		return nil, false
 	}
 
-	return &memberFound, true
+	return &found, true
 }
 
-// CheckRole checks if a member from a namespace can act over other with a specif role.
-func CheckRole(roleActive, rolePassive string) bool {
-	return authorizer.CheckRole(roleActive, rolePassive)
+// GetRoleCode converts a models.Member's role string to a role code. If the role is not found in Roles, it returns -1.
+func GetRoleCode(role string) int {
+	code, ok := Roles[role]
+	if !ok {
+		// return -1 when member role is not valid.
+		return -1
```

Comment:
As we are reworking this, we might consider using a type, even if it is an int behind the scenes, to have a more meaningful name when checking for it.

Reply:
Okay. It is a good an idea.

What do you think about a `InvalidRole` to map to this `-1`?